File: erubit.platform/src/main/java/a/erubit/platform/android/UserPresentBroadcastReceiver.java
package a.erubit.platform.android;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

public class UserPresentBroadcastReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {

        /*
         * Sent when the user is present after
         * device wakes up (e.g when the keyguard is gone)
         */
        if(Intent.ACTION_USER_PRESENT.equals(intent.getAction())) {
            showFloatingView(context);
        }
    }

    private void showFloatingView(Context context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
            context.startForegroundService(new Intent(context.getApplicationContext(), InteractionService.class));
        else
            context.startService(new Intent(context.getApplicationContext(), InteractionService.class));
    }
}

File: erubit.platform/src/main/java/a/erubit/platform/UserPresentBroadcastReceiver.kt
package a.erubit.platform

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build


class UserPresentBroadcastReceiver : BroadcastReceiver() {
	override fun onReceive(context: Context, intent: Intent) {
		/*
		 * Sent when the user is present after
		 * device wakes up (e.g when the keyguard is gone)
		 */

		if (Intent.ACTION_USER_PRESENT == intent.action) {
			showFloatingView(context)
		}
	}

	private fun showFloatingView(context: Context) {
		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
			context.startForegroundService(Intent(context.applicationContext, InteractionService::class.java))
		else
			context.startService(Intent(context.applicationContext, InteractionService::class.java))
	}
}
